main.go

```diff
@@ -79,19 +189,75 @@ func (b *Backend) healthCheck() {
 			fallthrough
 		case err == nil && resp.StatusCode == http.StatusOK:
 			resp.Body.Close()
-			if b.logging {
-				fmt.Printf("%s is up\n", b.endpoint)
+			var downtimeEnd time.Time
+			if (b.DowntimeStart != time.Time{}) {
+				now := time.Now().UTC()
+				downtime := now.Sub(b.DowntimeStart)
+				b.updateDowntime(downtime)
+				downtimeEnd = now
+			}
+			if globalLogFlag && !b.up && (b.Stats.UpSince != time.Time{}) {
```

Comment:
Define a global value `var timeZero = time.Time{}`  and then use that with `.Equal(timeZero)` helper